language-server/modules/langserver-compiler/src/test/java/org/ballerinalang/langserver/compiler/workspace/WorkspaceDocumentContentUpdateTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.langserver.compiler.workspace;
+
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Test document content update based on diff.
+ */
+public class WorkspaceDocumentContentUpdateTest {
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    public static final Path RES_DIR = Paths.get("src/test/resources/").toAbsolutePath();
+    private WorkspaceDocumentManagerImpl documentManager;
+
+    @BeforeClass
+    public void setUp() {
+        documentManager = WorkspaceDocumentManagerImpl.getInstance();
+        documentManager.clearAllFilePaths();
+    }
+
+    @Test(description = "Test the document range updating")
+    public void testDocumentRangeUpdate() throws IOException, WorkspaceDocumentException {
+        Path sourcePath1 = RES_DIR.resolve("source").resolve("docUpdateSource1.bal");
+        Path sourcePath2 = RES_DIR.resolve("source").resolve("docUpdateSource2.bal");
+        Path sourcePath3 = RES_DIR.resolve("source").resolve("docUpdateSource3.bal");
+        Path expected1Path = RES_DIR.resolve("expected").resolve("expected1.bal");
+        Path expected2Path = RES_DIR.resolve("expected").resolve("expected2.bal");
```

Comment:
can we these type of expected data to a data provider and then use a common logic to assert them ... ??

Reply:
@NipunaMarcus yes, There is a separate PR for optimizing the test with several other fixes.